central/deployment/datastore/datastore_test_constructors.go

```diff
@@ -0,0 +1,58 @@
+package datastore
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stackrox/rox/central/deployment/datastore/internal/search"
+	pgStore "github.com/stackrox/rox/central/deployment/datastore/internal/store/postgres"
+	imageDS "github.com/stackrox/rox/central/image/datastore"
+	nfDS "github.com/stackrox/rox/central/networkgraph/flow/datastore"
+	pbDS "github.com/stackrox/rox/central/processbaseline/datastore"
+	processIndicatorFilter "github.com/stackrox/rox/central/processindicator/filter"
+	"github.com/stackrox/rox/central/ranking"
+	riskDS "github.com/stackrox/rox/central/risk/datastore"
+	"github.com/stackrox/rox/pkg/expiringcache"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/process/filter"
+)
+
+// NewTestDataStore allows for direct creation of the datastore for testing purposes
+func NewTestDataStore(t testing.TB, testDB *pgtest.TestPostgres, images imageDS.DataStore, baselines pbDS.DataStore, networkFlows nfDS.ClusterDataStore, risks riskDS.DataStore, deletedDeploymentCache expiringcache.Cache, processFilter filter.Filter, clusterRanker *ranking.Ranker, nsRanker *ranking.Ranker, deploymentRanker *ranking.Ranker) (DataStore, error) {
```

Comment:
How about introducing a struct to hold all this datastores? Then when called we will nave named parameters so no need to check what is `nil`ed

Reply:
That sounds good :-)